### PR TITLE
fix MC-252817 (green map markers do not dissappear)

### PR DESCRIPTION
**Pull request by EmotionalLove**
It is a bug in Minecraft that green markers do not dissappear from maps when they are removed from Item Frame's. This patch fixes this issue. I hope I didn't screw up the formatting of my paper start/ends too badly.

### CODE REVIEW

**Comment by EmotionalLove** [comment]
https://bugs.mojang.com/browse/MC-252817 i intended to post this link with the original msg

File: patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
++            // paper end
+             MapItemSavedData worldmap = MapItem.getSavedData(i, this.level);
+ 
+             if (worldmap != null) {
+@@ -313,6 +317,20 @@ public class ItemFrame extends HangingEntity {
+         return OptionalInt.empty();
+     }
+ 
++    // paper start - fix MC-252817 (green map markers do not disappear)
++    public OptionalInt getMapIdFromItemStack(ItemStack stack) {
```
Comment (lynxplay) [inline patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch]: We can reduce the diff here a lot by delegating the entire `getFramedMapId` logic to this method by passing `this.getItem()` to it. E.g.
```patch
--- a/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
+++ b/src/main/java/net/minecraft/world/entity/decoration/ItemFrame.java
@@ -301,7 +301,12 @@ public class ItemFrame extends HangingEntity {

     public OptionalInt getFramedMapId() {
         ItemStack itemstack = this.getItem();
+    // Paper start - Fix remaining green markers (MC-252817)
+        return this.getFramedMapIdFromItem(itemstack);
+    }

+    public OptionalInt getFramedMapIdFromItem(final ItemStack itemstack) {
+    // Paper end - Fix remaining green markers (MC-252817)
         if (itemstack.is(Items.FILLED_MAP)) {
             Integer integer = MapItem.getMapId(itemstack);
```

Comment (EmotionalLove) [reply patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch]: I was thinking that, didn't get around to change it 'till now.

File: patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ 
+     public OptionalInt getFramedMapId() {
+         ItemStack itemstack = this.getItem();
++        // Paper start - fix MC-252817 (green map markers do not disappear)
++        return this.getFramedMapIdFromItem(itemstack);
++    }
+ 
+-        if (itemstack.is(Items.FILLED_MAP)) {
+-            Integer integer = MapItem.getMapId(itemstack);
++    public OptionalInt getFramedMapIdFromItem(ItemStack stack) {
```
Comment (lynxplay) [inline patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch]: If you rename the parameter here to `itemstack` the diff becomes even smaller :smile:

Comment (EmotionalLove) [reply patches/server/0936-fix-MC-252817-green-map-markers-do-not-disappear.patch]: you're right, i wasn't aware how much not changing the names of things actually affected the diff size till now haha

**Review by lynxplay — APPROVED** [review]
Generally looks good to me!
I'd suggest moving the detailed explanation of the bugs origin and how it is resolved into the patch subject (e.g. the commit's body when working in Paper-Server).

That way the code is a bit less cluttered.
Purely a quality of life change tho, your change works well in my testing :)

**Comment by EmotionalLove** [comment]
> Generally looks good to me! I'd suggest moving the detailed explanation of the bugs origin and how it is resolved into the patch subject (e.g. the commit's body when working in Paper-Server).
> 
> That way the code is a bit less cluttered. Purely a quality of life change tho, your change works well in my testing :)

I went ahead and did just that, much nicer. Thanks

**Review by Owen1212055 — APPROVED** [review]

**Comment by lynxplay** [comment]
As a heads up for hopefully following PRs, make sure to not open them on your projects master branch as it makes it harder for us to rebase your PR and merge it down the line :) Fine for this one tho, I hope we can get it merged today :+1:

**Review by electronicboy — APPROVED** [review]

**Pull request merged**